### PR TITLE
Engine: drop old internal comparision builtin

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
In this PR we change the internal for most of the comparison built-in function in oder to use the new value ordering.

This PR advances the state of #2256  

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [X] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [X] Include appropriate tests
- [X] Set a descriptive title and thorough description
- [X] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/svalue/Ordering.scala [hunk 1]
```diff
@@ -243,7 +243,10 @@ object Ordering extends scala.math.Ordering[SValue] {
             case SPAP(_, _, _) =>
               throw SErrorCrash("functions are not comparable")
           }
-        }(fallback = throw SErrorCrash("try to compare unrelated type"))
+        }(fallback = {
+          remy.log(tuple)
```
Comment (gerolf-da) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/svalue/Ordering.scala]: `remy.log`

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala [hunk 2]
```diff
@@ -322,8 +312,10 @@ final case class Compiler(packages: PackageId PartialFunction Package) {
               case BTextIntercalate => SBTextIntercalate
 
               // Implemented using normal SExpr
-              case BFoldl | BFoldr | BEqualList | BCoerceContractId | BTextMapEmpty |
-                  BGenMapEmpty =>
+              case BFoldl | BFoldr | BCoerceContractId | //
```
Comment (gerolf-da) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala]: Why the `//`s?

Comment (remyhaemmerle-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala]: trying to fight against auto-formatter.

dropped

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/svalue/Ordering.scala (hunk 1 above)
Comment (remyhaemmerle-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/svalue/Ordering.scala]: fixed

File: daml-lf/encoder/src/test/lf/Builtin_1.dev_.lf [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  val equal: forall (a: *). a -> a -> Bool =
+    EQUAL;
+  val less_eq: forall (a: *). a -> a -> Bool =
+    LESS_EQ;
+  val less_int64: forall (a: *). a -> a -> Bool =
+    LESS;
+  val greater_eq_int64: forall (a: *). a -> a -> Bool =
+    GREATER_EQ;
+  val greater_int64: forall (a: *). a -> a -> Bool =
+    GREATER_EQ;
```
Comment (gerolf-da) [inline daml-lf/encoder/src/test/lf/Builtin_1.dev_.lf]:
```suggestion
    GREATER;
```

File: daml-lf/encoder/src/test/lf/Builtin_1.7_1.8_.lf [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  val greater_eq_date: Date -> Date -> Bool =
+    GREATER_EQ @Date;
+  val greater_int64: Int64 -> Int64 -> Bool =
+    GREATER_EQ @Int64;
+  val greater_numeric: forall (n:nat). (Numeric n) -> (Numeric n) -> Bool =
+    GREATER_EQ_NUMERIC;
+  val greater_text: Text -> Text -> Bool =
+    GREATER @Text;
+  val greater_party: Party -> Party -> Bool =
+    GREATER_EQ @Party;
```
Comment (gerolf-da) [inline daml-lf/encoder/src/test/lf/Builtin_1.7_1.8_.lf]:
```suggestion
    GREATER @Party;
```

File: daml-lf/encoder/src/test/lf/Builtin_1.7_1.8_.lf [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+  val greater_eq: Text -> Text -> Bool =
+    GREATER_EQ @Text;
+  val greater_eq_party: Party -> Party -> Bool =
+    GREATER_EQ @Party;
+  val greater_eq_timestamp: Timestamp -> Timestamp -> Bool =
+    GREATER_EQ @Timestamp;
+  val greater_eq_date: Date -> Date -> Bool =
+    GREATER_EQ @Date;
+  val greater_int64: Int64 -> Int64 -> Bool =
+    GREATER_EQ @Int64;
```
Comment (gerolf-da) [inline daml-lf/encoder/src/test/lf/Builtin_1.7_1.8_.lf]:
```suggestion
    GREATER @Int64;
```

File: daml-lf/encoder/src/test/lf/Builtin_1.7_1.8_.lf [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+  val greater_numeric: forall (n:nat). (Numeric n) -> (Numeric n) -> Bool =
+    GREATER_EQ_NUMERIC;
```
Comment (gerolf-da) [inline daml-lf/encoder/src/test/lf/Builtin_1.7_1.8_.lf]:
```suggestion
    GREATER_NUMERIC;
```

File: daml-lf/encoder/src/test/lf/Builtin_1.dev_.lf [hunk 7]
```diff
@@ -0,0 +1,17 @@
+// Copyright (c) 2020 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+module BuiltinMod {
+
+  val equal: forall (a: *). a -> a -> Bool =
+    EQUAL;
+  val less_eq: forall (a: *). a -> a -> Bool =
+    LESS_EQ;
+  val less_int64: forall (a: *). a -> a -> Bool =
```
Comment (gerolf-da) [inline daml-lf/encoder/src/test/lf/Builtin_1.dev_.lf]:
```suggestion
  val less: forall (a: *). a -> a -> Bool =
```

File: daml-lf/encoder/src/test/lf/Builtin_1.dev_.lf [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+    LESS;
+  val greater_eq_int64: forall (a: *). a -> a -> Bool =
```
Comment (gerolf-da) [inline daml-lf/encoder/src/test/lf/Builtin_1.dev_.lf]:
```suggestion
  val greater_eq: forall (a: *). a -> a -> Bool =
```

File: daml-lf/encoder/src/test/lf/Builtin_1.dev_.lf [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+    GREATER_EQ;
+  val greater_int64: forall (a: *). a -> a -> Bool =
```
Comment (gerolf-da) [inline daml-lf/encoder/src/test/lf/Builtin_1.dev_.lf]:
```suggestion
  val greater: forall (a: *). a -> a -> Bool =
```

File: daml-lf/encoder/src/test/lf/Builtin_1.1_1.3_1.6_.lf [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+  val greater_eq_date: Date -> Date -> Bool =
+    GREATER_EQ @Date;
+  val greater_int64: Int64 -> Int64 -> Bool =
+    GREATER_EQ @Int64;
+  val greater_decimal: (Numeric 10) -> (Numeric 10) -> Bool =
+    GREATER_EQ @(Numeric 10);
+  val greater_text: Text -> Text -> Bool =
+    GREATER @Text;
+  val greater_party: Party -> Party -> Bool =
+    GREATER_EQ @Party;
```
Comment (gerolf-da) [inline daml-lf/encoder/src/test/lf/Builtin_1.1_1.3_1.6_.lf]:
```suggestion
    GREATER @Party;
```

File: daml-lf/encoder/src/test/lf/Builtin_1.1_1.3_1.6_.lf [hunk 11]
```diff
@@ -0,0 +1,67 @@
...
+  val greater_eq_party: Party -> Party -> Bool =
+    GREATER_EQ @Party;
+  val greater_eq_timestamp: Timestamp -> Timestamp -> Bool =
+    GREATER_EQ @Timestamp;
+  val greater_eq_date: Date -> Date -> Bool =
+    GREATER_EQ @Date;
+  val greater_int64: Int64 -> Int64 -> Bool =
+    GREATER_EQ @Int64;
+  val greater_decimal: (Numeric 10) -> (Numeric 10) -> Bool =
+    GREATER_EQ @(Numeric 10);
```
Comment (gerolf-da) [inline daml-lf/encoder/src/test/lf/Builtin_1.1_1.3_1.6_.lf]:
```suggestion
    GREATER @(Numeric 10);
```

File: daml-lf/encoder/src/test/lf/Builtin_1.1_1.3_1.6_.lf [hunk 12]
```diff
@@ -0,0 +1,67 @@
...
+  val greater_eq: Text -> Text -> Bool =
+    GREATER_EQ @Text;
+  val greater_eq_party: Party -> Party -> Bool =
+    GREATER_EQ @Party;
+  val greater_eq_timestamp: Timestamp -> Timestamp -> Bool =
+    GREATER_EQ @Timestamp;
+  val greater_eq_date: Date -> Date -> Bool =
+    GREATER_EQ @Date;
+  val greater_int64: Int64 -> Int64 -> Bool =
+    GREATER_EQ @Int64;
```
Comment (gerolf-da) [inline daml-lf/encoder/src/test/lf/Builtin_1.1_1.3_1.6_.lf]:
```suggestion
    GREATER @Int64;
```

File: daml-lf/encoder/src/test/lf/Builtin_1.0_.lf [hunk 13]
```diff
@@ -0,0 +1,59 @@
...
+  val greater_eq_decimal: (Numeric 10) -> (Numeric 10) -> Bool =
+    GREATER_EQ @(Numeric 10);
+  val greater_eq: Text -> Text -> Bool =
+    GREATER_EQ @Text;
+  val greater_eq_timestamp: Timestamp -> Timestamp -> Bool =
+    GREATER_EQ @Timestamp;
+  val greater_eq_date: Date -> Date -> Bool =
+    GREATER_EQ @Date;
+  val greater_int64: Int64 -> Int64 -> Bool =
+    GREATER_EQ @Int64;
```
Comment (gerolf-da) [inline daml-lf/encoder/src/test/lf/Builtin_1.0_.lf]:
```suggestion
    GREATER @Int64;
```

File: daml-lf/encoder/src/test/lf/Builtin_1.0_.lf [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+  val greater_decimal: (Numeric 10) -> (Numeric 10) -> Bool =
+    GREATER_EQ @(Numeric 10);
```
Comment (gerolf-da) [inline daml-lf/encoder/src/test/lf/Builtin_1.0_.lf]:
```suggestion
    GREATER @(Numeric 10);
```

File: daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala [hunk 15]
```diff
@@ -194,6 +194,17 @@ final case class Compiler(packages: PackageId PartialFunction Package) {
   private val compileGetTime: SExpr =
     SEAbs(1) { SBGetTime(SEVar(1)) }
 
+  private val SBLessNumeric =
+    SEAbs(3, SEApp(SEBuiltin(SBLess), Array(SEVar(2), SEVar(1))))
```
Comment (gerolf-da) [inline daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala]: Is this to "drop" one of the parameters?

Comment (remyhaemmerle-da) [reply daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala]: This drop the type parameter `nat`, which is nor erase.

**Pull request merged**